pkg/cmd/cli/secrets/source.go

```diff
@@ -0,0 +1,209 @@
+)
+
+func NewCmdCreateSourceSecret(name, fullName string, f *kcmdutil.Factory, reader io.Reader, out io.Writer, newSecretFullName, ocEditFullName string) *cobra.Command {
+	o := &CreateSourceSecretOptions{
+		Out:    out,
+		Reader: reader,
+	}
+
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s SECRET_NAME --username=USERNAME --password=PASSWORD --ca-cert=CERTIFICATE --ssh-private-key=SSH_KEY --gitconfig=GITCONFIG", name),
+		Short: "Create a new source secret",
+		Long:  fmt.Sprintf(createSourceSecretLong, fullName, newSecretFullName, ocEditFullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := o.Complete(f, args); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.Validate(); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.CreateSourceSecret(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&o.Username, "username", "", "username for Git authentication")
+	cmd.Flags().StringVar(&o.Password, "password", "", "password or token for Git authentication")
+	cmd.Flags().StringVar(&o.PrivateKeyPath, "ssh-privatekey", "", "path to the ssh private key")
+	cmd.Flags().StringVar(&o.CertificatePath, "cert-path", "", "path to the certificate file")
+	cmd.Flags().StringVar(&o.GitConfigPath, "gitconfig-path", "", "path to the .gitconfig file")
+	cmd.Flags().BoolVarP(&o.PromptForPassword, "prompt", "", false, "Prompt for password or token")
+	kcmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+func (o *CreateSourceSecretOptions) CreateSourceSecret() error {
+	secret, err := o.MakeSourceSecret()
+	if err != nil {
+		return err
+	}
+
+	if _, err := o.SecretsInterface.Create(secret); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *CreateSourceSecretOptions) MakeSourceSecret() (*api.Secret, error) {
+
+	secret := &api.Secret{}
+	secret.Namespace = o.SecretNamespace
+	secret.Name = o.SecretName
+	secret.Type = SecretTypeSource
+	secret.Data = map[string][]byte{}
+
+	if len(o.Username) != 0 {
+		secret.Data["username"] = []byte(o.Username)
+	}
+
+	if len(o.Password) != 0 {
+		secret.Data["password"] = []byte(o.Password)
+	}
+
+	if len(o.CertificatePath) != 0 {
+		caContent, err := ioutil.ReadFile(o.CertificatePath)
+		if err != nil {
+			return nil, err
+		}
+		secret.Data["ca.crt"] = caContent
+	}
+
+	if len(o.GitConfigPath) != 0 {
+		gitConfig, err := ioutil.ReadFile(o.GitConfigPath)
+		if err != nil {
+			return nil, err
+		}
+		secret.Data["gitconfig"] = gitConfig
+	}
+
+	if len(o.PrivateKeyPath) != 0 {
+		privateKeyContent, err := ioutil.ReadFile(o.PrivateKeyPath)
+		if err != nil {
+			return nil, err
+		}
+		secret.Data["ssh-privatekey"] = privateKeyContent
+	}
+
+	return secret, nil
+}
+
+func (o *CreateSourceSecretOptions) Complete(f *kcmdutil.Factory, args []string) error {
+
+	if o.PromptForPassword {
+		if cmdutil.IsTerminal(o.Reader) {
+			for !o.passwordProvided() {
```

Comment:
I would think that if you have the flag to prompt for the password, you always prompt regardless of a password being provided. Also, why 'for' ?